### DISCUSSION TITLE
VRL return early

### DISCUSSION THREAD

**Discussion by arve0** [vectordotdev/vector#16098] [Q&A] [answered]
Hi! I see that `return` is a reserved keyword:

> error[E205]: reserved keyword
>    ┌─ :56:3
>    │
> 56 │   return
>    │   ^^^^^^
>    │   │
>    │   this identifier name is reserved for future use in the language
>    │   use a different name instead
>    │
>    = see language documentation at https://vrl.dev
>    = try your code in the VRL REPL, learn more at https://vrl.dev/examples

What are people doing instead of returning early? Using several transforms, with some condition?

I find the nested `if err != null { if err != null { ....` a bit annoying.

**Comment by arve0** [comment, marked as answer]
I see `abort` with [drop_on_abort](https://vector.dev/docs/reference/configuration/transforms/remap/#drop_on_abort) and [reroute_dropped](https://vector.dev/docs/reference/configuration/transforms/remap/#reroute_dropped) gives some flexibility.

**Reply by davidhuie-dd** [reply]
Yeah, you can use the abort features. 

Another thing to try is to implement routing, where you can write VRL for specific types of logs: https://vector.dev/docs/reference/configuration/transforms/route/#examples
